src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -233,6 +235,25 @@ void Complete ()
 			foreach (var body in _unreachableBodies) {
 				Annotations.SetAction (body.Method, MethodAction.ConvertToThrow);
 			}
+
+			foreach (var attr in _ivt_attributes) {
+				if (IsInternalsVisibleAttributeAssemblyMarked (attr.Attribute))
+					MarkCustomAttribute (attr.Attribute, new DependencyInfo (DependencyKind.AssemblyOrModuleAttribute, attr.Provider), null);
+			}
+
+			bool IsInternalsVisibleAttributeAssemblyMarked (CustomAttribute ca)
+			{
+				var assemblyName = (string) ca.ConstructorArguments[0].Value;
+				int comma = assemblyName.IndexOf (',');
```

Comment:
Would it be better to roundtrip this through `AssemblyName`? This string has various rules that need to be followed (whitespace is allowed and gets trimmed, the name can be enclosed in quotation marks, etc.).

Reply:
Good idea.